Runtime/Shapes/ShapeComponent.cs

```diff
@@ -0,0 +1,144 @@
+﻿using System;
+
+namespace UnityEngine.ProBuilder
+{
+    [RequireComponent(typeof(ProBuilderMesh))]
+    public class ShapeComponent : MonoBehaviour
+    {
+        [SerializeReference]
+        public Shape shape = new Cube();
+
+        public bool isInit;
```

Comment:
noted above, let's see if we can figure out a way to avoid needing an init flag in this class